llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -9248,20 +9249,23 @@ calculateByteProvider(SDValue Op, unsigned Index, unsigned Depth,
     if (!L->isSimple() || L->isIndexed())
       return std::nullopt;
 
-    unsigned NarrowBitWidth = L->getMemoryVT().getSizeInBits();
-    if (NarrowBitWidth % 8 != 0)
+    TypeSize NarrowBitWidth = L->getMemoryVT().getSizeInBits();
```

Comment:
Should this be getScalarSizeInBits? The code on line 9260 is trying to figure out if it is in the zero extended part of the load. That should be based on the scalar size not the full vector size.

Reply:
Yes, I guess I've taken the code too literally.  As long as `Index` is the byte index within a scalar, which looks to be the code based on the existing call sites, then I agree this should be using the MemoryVT's scalar size.

Reply:
Fixed.